interface-spec.md

```diff
@@ -108,11 +108,17 @@ Triple is an alias of Quad.
 
 ### DataFactory
 
+For default values of the instance properties, see the individual [interface
+definitions](#data-interfaces)
+
 **Methods:**
 
 - `IRI .iri(String iri)` returns a new instance of IRI.
 - `BlankNode .blankNode()` returns a new instance of BlankNode.
-- `Literal .literal(String value, String language, String datatype)` returns a new instance of Literal.
+- Literal .literal(String value, [String languageOrDatatype]) returns a new
+  instance of Literal. If languageOrDatatype is an IRI, then a NamedNode is
```

Comment:
The factory already allowed string values for the datatype. If it's a string the function must distinguish between languages and IRIs. The logic of the method would be something like this:

    factory.literal = function (value, languageOrDatatype) {
      if (typeof languageOrDatatype === 'string') {
        // RegExp magic
        if (isLanguage(languageOrDatatype)) {
          return new Literal(value, languageOrDatatype, langString)
        } else {
          return new Literal(value, '', this.NamedNode(languageOrDatatype))
        }
      } else {
        // superfast without detection
        return new Literal(value, '', languageOrDatatype)
      }
    }

Is it possible to implement the `isLanguage()` function in a save way? The RDF spec points to [BCP 47 2.2.9](http://tools.ietf.org/html/bcp47#section-2.2.9). Is this: `[a-z]{1,8}-[a-z]{1,8}` enough? Could this cause a conflict with an IRI?